keel-plugin/src/main/kotlin/com/netflix/spinnaker/keel/plugin/ResolvableResourceHandler.kt

```diff
@@ -0,0 +1,161 @@
+package com.netflix.spinnaker.keel.plugin
+
+import com.fasterxml.jackson.databind.ObjectMapper
+import com.netflix.spinnaker.keel.api.ApiVersion
+import com.netflix.spinnaker.keel.api.Resource
+import com.netflix.spinnaker.keel.api.ResourceKind
+import com.netflix.spinnaker.keel.api.ResourceMetadata
+import com.netflix.spinnaker.keel.api.ResourceName
+import com.netflix.spinnaker.keel.api.SubmittedResource
+import com.netflix.spinnaker.keel.api.randomUID
+import com.netflix.spinnaker.keel.events.TaskRef
+import com.netflix.spinnaker.keel.exceptions.InvalidResourceStructureException
+import org.slf4j.Logger
+
+/**
+ * @param S the spec type.
+ * @param R the resolved model type.
+ *
+ * If those two are the same, use [ResourceHandler] instead.
+ */
+interface ResolvableResourceHandler<S : Any, R : Any> : KeelPlugin {
```

Comment:
Super-interface of `ResourceHandler` where the spec and desired/current types may be different